verification/cbmc/sources/utils.c

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <utils.h>
+
+void assert_byte_buf_contents_are_equal(const struct aws_byte_buf *const lhs, const struct aws_byte_buf *const rhs) {
+    assert(lhs && rhs);
+    assert(lhs->len == rhs->len);
+    if (lhs->len > 0) {
+        size_t index;
+        __CPROVER_assume(index < lhs->len);
+        assert(lhs->buffer[index] == rhs->buffer[index]);
+    }
+}
```

Comment:
@adpaco-aws You should create a GitHub issue to move this function to AWS C Common. All utility function over AWS C Common data structures should go to [utils.c](https://github.com/awslabs/aws-c-common/blob/main/verification/cbmc/sources/utils.c). Even better, I'd suggest you to write it as
```c
bool aws_byte_buf_contents_match(const struct aws_byte_buf *const lhs, const struct aws_byte_buf *const rhs) {
    if((!lhs || !rhs)) /* Filter null pointers. */
        return (lhs == rhs); /* Return true if both null. */
    if(lhs->len != rhs->len)
        return false;
    if(lhs->len > 0) {
        size_t index;
        __CPROVER_assume(index < lhs->len);
        return (lhs->buffer[index] == rhs->buffer[index]);
    }
    return true;
}
```

That way you could use for both equality
```c
assert(aws_byte_buf_contents_match(lhs, rhs));
```
and inequality
```c
assert(!aws_byte_buf_contents_match(lhs, rhs));
```
Same for `assert_keys_are_equal` below. What do you think, @alex-chew ?